### PR TITLE
[SPARK-8525][MLLIB] fix LabeledPoint parser when there is a whitespace between label and features vector

### PR DESCRIPTION
**Pull request by fe2s**
fix LabeledPoint parser when there is a whitespace between label and features vector, e.g.
(y, [x1, x2, x3])

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/util/NumericParser.scala [hunk 1]
```diff
@@ -19,6 +19,8 @@ package org.apache.spark.mllib.util
 
 import java.util.StringTokenizer
 
+import org.apache.commons.lang.StringUtils.isBlank
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/util/NumericParser.scala]: Don't use commons-lang -- it's not actually a dependency. I wouldn't even use lang3. You don't need a utility class for something this simple

File: mllib/src/test/scala/org/apache/spark/mllib/regression/LabeledPointSuite.scala [hunk 2]
```diff
@@ -31,6 +31,11 @@ class LabeledPointSuite extends SparkFunSuite {
     }
   }
 
+  test("parse labeled points with whitespaces") {
+    val point = LabeledPoint.parse("(0.0, [1.0, 2.0])")
```
Comment (srowen) [inline mllib/src/test/scala/org/apache/spark/mllib/regression/LabeledPointSuite.scala]: Hm, doesn't this still yield ` 2.0` as a token during parsing? which wouldn't parse I'd imagine. Does this pass? in which case I'm missing something ...

**Comment by srowen** [comment]
CC @mengxr to see if whitespace is supposed to be tolerated

File: mllib/src/test/scala/org/apache/spark/mllib/regression/LabeledPointSuite.scala (hunk 2 above)
Comment (fe2s) [inline mllib/src/test/scala/org/apache/spark/mllib/regression/LabeledPointSuite.scala]: When it comes to java.lang.Double.parseDouble(s) it ignores spaces.

Comment (srowen) [inline mllib/src/test/scala/org/apache/spark/mllib/regression/LabeledPointSuite.scala]: Oh, interesting. Sounds good.

**Comment by mengxr** [comment]
LGTM. Merged into master. Though the format is designed for internal save/load only, it is still good to have some robustness. Also note that the preferred format is Parquet now. Thanks!

**Pull request closed without merging**